paddle/fluid/memory/memcpy.cc

```diff
@@ -70,6 +78,9 @@ void Copy<platform::CPUPlace, platform::CUDAPlace>(
       SyncCUDAStream();
     }
   }
+  if (change_device) {
+    platform::SetDeviceId(device_id);
```

Comment:
I don't see why this is necessary, as  `CUDADeviceContext` would switch to proper  device for next operation.